src/concurrency/decentralized_epoch_manager.cpp

```diff
@@ -18,33 +18,34 @@ namespace concurrency {
 
 
   // enter epoch with thread id
-  cid_t DecentralizedEpochManager::EnterEpoch(const size_t thread_id) {
+  cid_t DecentralizedEpochManager::EnterEpoch(const size_t thread_id, const bool is_snapshot_read) {
```

Comment:
both. Whenever a transaction is executed under snapshot isolation or declared as read-only, it will directly access a snapshot version.